build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/DockerPush.java

```diff
@@ -103,19 +92,43 @@ public Build execute() {
         try {
             DockerJavaWrapper.pushImage(imageTag, username, password, host, env, logger);
             String imageId = DockerJavaWrapper.getImageIdFromTag(imageTag, host, env, logger);
-            DockerImage image = new DockerImage(imageId, imageTag, deploymentRepository);
-            Module module = image.generateBuildInfoModule(buildInfoClientBuilder, dependenciesClientBuilder, logger, artifactProperties);
+            DockerImage image = new DockerImage(imageId, imageTag, targetRepository, buildInfoClientBuilder, dependenciesClientBuilder, "", "");
+            Module module = image.generateBuildInfoModule(logger, DockerUtils.CommandType.Push);
             if (module.getArtifacts() == null || module.getArtifacts().size() == 0) {
                 logger.warn("Could not find docker image: " + imageTag + " in Artifactory.");
             }
+            setImageLayersProps(image.getLayers(), artifactProperties, buildInfoClientBuilder);
             Build build = new Build();
             modulesList.add(module);
             build.setModules(modulesList);
             logger.info("Successfully pushed docker image: " + imageTag);
             return build;
-        } catch (IOException e) {
+        } catch (IOException | InterruptedException e) {
             logger.error(e.getMessage(), e);
             throw new RuntimeException(e);
         }
     }
-}
+
+    /**
+     * Update each layer's properties with artifactProperties.
+     */
+    private void setImageLayersProps(DockerLayers layers, ArrayListMultimap<String, String> artifactProperties, ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder) throws IOException {
+        String artifactsPropsStr = buildMatrixParamsString(artifactProperties, false);
+        try (ArtifactoryBuildInfoClient buildInfoClient = buildInfoClientBuilder.build()) {
+            for (DockerLayer layer : layers.getLayers()) {
+                HttpResponse httpResponse = buildInfoClient.executeUpdateFileProperty(layer.getFullPath(), artifactsPropsStr);
+                validateSetImageLayersResponse(httpResponse);
+                EntityUtils.consume(httpResponse.getEntity());
+
```

Comment:
Lets make sure we always consume the HTTP response:
```suggestion
                HttpResponse httpResponse = null;
                try {
                    httpResponse = buildInfoClient.executeUpdateFileProperty(layer.getFullPath(), artifactsPropsStr);
                    validateSetImageLayersResponse(httpResponse);
                } finally {
                    if (httpResponse != null) {
                        EntityUtils.consume(httpResponse.getEntity());
                    }
                }
```